deepcell/training/model_trainer.py

```diff
@@ -0,0 +1,486 @@
+                seed = self.generator_seed,
+                batch_size = self.generator_batch_size,
+                transform = self.generator_transform,
+                transform_kwargs = self.generator_transform_kwargs)
+        ## validation image generator
+        validation_dict = {"X": self.X_test, "y": self.y_test}
+        self.validation_data, self.output_metadata = prep_generator(
+                generator_type = "validation",
+                output_metadata = self.output_metadata,
+                data_generator = self.validation_generator,
+                input_data_dict = validation_dict,
+                skip = self.generator_skips,
+                seed = self.generator_seed,
+                batch_size = self.generator_batch_size,
+                transform = self.generator_transform,
+                transform_kwargs = self.generator_transform_kwargs)
+        print(self.output_metadata)
+
+    def _compile_model(self):
+        
+        def model_compilation(
+                output_metadata,
+                model,
+                loss_function,
+                optimizer,
+                optimizer_flag, 
+                optimizer_learning_rate,
+                optimizer_decay,
+                optimizer_momentum,
+                optimizer_nesterov,
+                metrics):
+            model.compile(loss=loss_function, optimizer=optimizer, metrics=metrics)
+            output_metadata["model"] = {}
+            output_metadata["model"]["compilation"] = {}
+            output_metadata["model"]["compilation"]["loss_function"] = loss_function
+            if optimizer_flag: # optimizer was passed in by user
+                output_metadata["model"]["compilation"]["optimizer"] = "need more information; can't access initialization parameters"
+            else: # optimizer is set to default value
+                output_metadata["model"]["compilation"]["optimizer"] = {}
+                output_metadata["model"]["compilation"]["optimizer"]["class"] = re.match("<([\w.]+) object at",str(optimizer)).group(1)
+                output_metadata["model"]["compilation"]["optimizer"]["learning_rate"] = optimizer_learning_rate
+                output_metadata["model"]["compilation"]["optimizer"]["decay"] = optimizer_decay
+                output_metadata["model"]["compilation"]["optimizer"]["momentum"] = optimizer_momentum
+                output_metadata["model"]["compilation"]["optimizer"]["nesterov"] = optimizer_nesterov
+            output_metadata["model"]["compilation"]["metrics"] = metrics
+
+        metrics = ["accuracy"]
+        model_compilation(
+                self.output_metadata,
+                self.model,
+                self.loss_function,
+                self.optimizer,
+                self.optimizer_passed,
+                self.optimizer_learning_rate,
+                self.optimizer_decay,
+                self.optimizer_momentum,
+                self.optimizer_nesterov,
+                metrics)
```

Comment:
I'm not sure I understand the value of the `model_compilation` function. why add this layer?